.buildkite/scripts/steps/build-agent-core.sh

```diff
@@ -2,7 +2,9 @@
 
 set -euo pipefail
 
-source .buildkite/scripts/bootstrap.sh
+source ../scripts/common.sh
+BEAT_VERSION=$(grep -oE '[0-9]+\.[0-9]+\.[0-9]+(\-[a-zA-Z]+[0-9]+)?' "${WORKSPACE}/version/version.go")
```

Comment:
This can be simplified (there are no `-SNAPSHOT` or other suffixes in version.go, those come from the packaging process): 
```suggestion
BEAT_VERSION=$(grep -oE '[0-9]+\.[0-9]+\.[0-9]+' "${WORKSPACE}/version/version.go")
```